src/Swarm/TUI/Controller.hs

```diff
@@ -406,11 +407,18 @@ handleModalEvent = \case
     Brick.zoom (uiState . uiModal . _Just . modalDialog) (handleDialogEvent ev)
     modal <- preuse $ uiState . uiModal . _Just . modalType
     case modal of
-      Just GoalModal -> do
-        lw <- use $ uiState . uiGoal . listWidget
-        newList <- refreshList lw
-        uiState . uiGoal . listWidget .= newList
-      Just _ -> handleInfoPanelEvent modalScroll (VtyEvent ev)
+      Just GoalModal -> case ev of
+        V.EvKey (V.KChar '\t') [] -> uiState . uiGoal . focus %= focusNext
+        _ -> do
+          focused <- use $ uiState . uiGoal . focus
+          case focusGetCurrent focused of
+            Just (GoalWidgets w) -> case w of
+              ObjectivesList -> do
+                lw <- use $ uiState . uiGoal . listWidget
+                newList <- refreshList lw
+                uiState . uiGoal . listWidget .= newList
+              GoalSummary -> handleInfoPanelEvent modalScroll (VtyEvent ev)
```

Comment:
The strange thing now is that the entire modal is one big scrollable widget, so when you focus on the description you can scroll both the description and the objective list together; and conversely when you are focused on the objective list it doesn't scroll, even if the currently highlighted objective is off the bottom of the modal.  I would think that (1) the objective list and the description should be made scrollable separately (maybe the objective list has a scrollbar on the left?), and (2) navigating through the objective list should cause it to scroll just like we do for inventory.  I don't quite remember how that works for the inventory but I don't think it's hard, it's mostly handled automatically by brick.

Reply:
Fixed now.  List widgets do have scrolling built in (without scroll bars).